src/coreclr/jit/lower.cpp

```diff
@@ -1915,7 +1915,10 @@ GenTree* Lowering::LowerCallMemcmp(GenTreeCall* call)
             {
                 MaxUnrollSize = 64;
             }
-// TODO-XARCH-AVX512: Consider enabling this for AVX512
+            if (comp->compOpportunisticallyDependsOn(InstructionSet_Vector512))
+            {
+                MaxUnrollSize = 128;
+            }
```

Comment:
nit: Put this first and make `Vector256` an `else`, that way we aren't doing unnecessary extra checks.